Subject: [PATCH] SystemUI: Fix status bar tuner crash on rotation

The TunerFragment that SubSettingsFragment holds a reference
to gets destroyed, causing the settings to crash upon rotation.
Set the target fragment to null after we're done using it.

diff --git a/packages/SystemUI/src/com/android/systemui/tuner/TunerActivity.java b/packages/SystemUI/src/com/android/systemui/tuner/TunerActivity.java
index cd6474e..9981d95 100644
--- a/packages/SystemUI/src/com/android/systemui/tuner/TunerActivity.java
+++ b/packages/SystemUI/src/com/android/systemui/tuner/TunerActivity.java
@@ -127,10 +127,14 @@
     public static class SubSettingsFragment extends PreferenceFragment {
         @Override
         public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+            if (getTargetFragment() == null) {
+                return;
+            }
             PreferenceScreen p = (PreferenceScreen) ((PreferenceFragment) getTargetFragment())
                     .getPreferenceScreen().findPreference(rootKey);
             setPreferenceScreen(p);
             getActivity().getActionBar().setTitle(p.getTitle());
+            setTargetFragment(null, 0);
         }
     }
 }
